contracts/modules/RarityShuffleMetadata.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import { LibString } from "solady/utils/LibString.sol";
+import { ISoundEditionV1a } from "@core/interfaces/ISoundEditionV1a.sol";
+import { IRarityShuffleMetadata } from "@modules/interfaces/IRarityShuffleMetadata.sol";
+
+/**
+ * @title RarityShuffleMetadata
+ * @dev Module for shuffling each tokenID of a Sound protocol mint and bucketing shuffled IDs into rarity tiers
+ * @author @ipatka
+ */
+contract RarityShuffleMetadata is IRarityShuffleMetadata {
+    mapping(uint256 => uint16) public availableIds; /*Mapping to use to track used offsets*/
+    uint16 public availableCount; /*Track the available count to know the id of the current max offset*/
+    mapping(uint256 => uint256) public offsets; /*Store offsets once found*/
+
+    uint256 public nextIndex; /*Store next index to set*/
+
+    uint256[] public ranges; /*Store range sizes for song Ids*/
+    uint256 public nRanges; /*Store number of rarity tiers*/
+
+    address public edition; /*The address of the edition that can trigger metadata updates*/
```

Comment:
If you change nextIndex and nRanges to uint32 (supports up to 4294967296)
You can reorder this so 
```
address public edition;
uint32 public nRanges;
uint32 public nextIndex;
uint16 public availableCount;
```

will fit into one storage slot and reduce a few slots worth of gas

https://fravoll.github.io/solidity-patterns/tight_variable_packing.html

Reply:
Thanks will update